### PR TITLE
KAFKA-5644: Fix Reset Consumer Group Offset tool to handle minute component of TimeZone

### PR DESCRIPTION
**Pull request by omkreddy**

### CODE REVIEW

**Comment by omkreddy** [comment]
ResetConsumerGroupOffsetTest.testResetOffsetsToZonedDateTime is failing on my computer which is on IST timezone (+5:30). Since we are not handling minute component of timezone it was failing.
 cc @ijuma

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6581/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6566/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by baluchicken** [comment]
LGTM:) Thanks for the patch.

**Comment by ijuma** [comment]
cc @hachikuji

**Comment by omkreddy** [comment]
@hachikuji pinging for review

**Comment by hachikuji** [comment]
@omkreddy Looks good. Could we get a test case for this?

**Comment by omkreddy** [comment]
@hachikuji I already updated existing test case. This should work in all time zones. Do you want any specific test case?

**Comment by hachikuji** [comment]
@omkreddy Hmm i see.. Was the old format incorrect or just incomplete? In other words, would a date formatted with the old format string still parse correctly?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6765/
Test FAILed (JDK 8 and Scala 2.12).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6767/
Test FAILed (JDK 8 and Scala 2.12).

**Comment by omkreddy** [comment]
@hachikuji yes, the old format was bit incomplete.  While parsing, it was not taking the minute part of the time zone offsets.  Updated code to fall back to the old format for any parsing errors.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6783/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6768/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by hachikuji** [comment]
@omkreddy Since we have the path to handle the old format string (by catching `ParseException`), we should probably have a test case for it as well.

**Comment by omkreddy** [comment]
@hachikuji updated the test case to test old format. Since old format does not work on some time zones, it just checks the code path.

**Comment by omkreddy** [comment]
cc @jeqo Can you also check this PR?

File: core/src/test/scala/unit/kafka/admin/ResetConsumerGroupOffsetTest.scala [hunk 1]
```diff
@@ -174,24 +178,31 @@ class ResetConsumerGroupOffsetTest extends KafkaServerTestHarness {
...
     }, "Expected the consumer group to reset to when offset was 50.")
 
     printConsumerGroup()
 
+    // test old format. don't do validity check, since it wont work on some timezones
+    val format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX")
+    val cgcArgs2 = Array("--bootstrap-server", brokerList, "--reset-offsets", "--group", group, "--all-topics", "--to-datetime", format.format(checkpoint), "--execute")
+    val opts2 = new ConsumerGroupCommandOptions(cgcArgs2)
+    val consumerGroupCommand2 = createConsumerGroupService(opts2)
+    consumerGroupCommand2.resetOffsets()
```
Comment (hachikuji) [inline core/src/test/scala/unit/kafka/admin/ResetConsumerGroupOffsetTest.scala]: Not sure what we're actually validating here. Maybe we could just test the `getDateTime` function directly in a separate test case to ensure compatibility?

**Comment by omkreddy** [comment]
@hachikuji  Thanks for the review.  added a test-case to check supported date time formats.

some cleanups were done.
1. changed the package util.kaka.admin to kafka.admin
2. with above change, there were some errors due to redundant classes (ConsumerThread, ConsumerGroupExecutor) in both  ResetConsumerGroupOffsetTest, DescribeConsumerGroupTest tests.  removed these redundant internal classes from  ResetConsumerGroupOffsetTest.

**Review by hachikuji — APPROVED** [review]
LGTM. Really appreciate the extra cleanup!

**Comment by jeqo** [comment]
LGTM also, thanks @omkreddy ! :+1:

**Pull request closed without merging**